docs/source/blocks/editcomponent.md

```diff
@@ -127,6 +127,43 @@ import { Icon } from '@plone/volto/components';
 </SidebarPortal>;
 ```
 
+### Default values support for block schema forms
+
+In the block schema definition you can use the two additional properties for
+each field: `default` and `missing`. The `default` value is used to populate
+the data as "initial value". The `missing` value should be used when the field
+should always have a value (as required).
+
+```
+  // ...
+  properties: {
+    url: {
+      title: 'Embed URL',
+      default: 'https://example.com',
+      missing: 'https://nothing.com',
+    },
+  }
+  // ...
+```
+
+In effect, the following situation matrix will occur:
+
+- `default` and `missing` are declared: the block form is initialized with
+  the `default` value. When the user clears the field, the field is
+  reinitialized with the `missing` value. This would imply a "required" field.
+- only `default` is declared: the block form is initialized with the `default`
+  value. When the user clears the field, the field remains empty. This would
+  imply an "optional" field.
+- only `missing` is declared: the block form is initialized with the `missing`
+  value. When the user clears the field, the field is reinitialized with the
+  `missing` value.
```

Comment:
I'm curious. Does this follow the same pattern as Colander, which shares its roots with Zope? https://docs.pylonsproject.org/projects/colander/en/latest/api.html#schema-related

We need to add the case when neither `missing` nor `default` is defined.

The scenarios also need to be clarified about when reinitialization occurs (when field is cleared or form submitted) and where (client or server side).

Reply:
I think I'm more used to the zope.schema concept, rather then colander.

https://github.com/zopefoundation/zope.schema/blob/89f09bb970bc14f81b724e81174aaaedb8853c3a/src/zope/schema/interfaces.py#L251-L261C9

The major difference to the zope.schema is that the UI is automatically updated based on the schema and user interactions, so it's more a "user interaction" issue rather then data validation

Reply:
OK, that makes a little more sense, I guess kinda like HTML inputs getting their attributes for `value` and `required` set and unset?

I would suggest the following, and I have further questions to help me understand and clarify the intent.

Let's make this section a definition list for ease of legibility.

```
`default` and `missing` are declared
:   The block form is initialized with the `default` value.
    When the user clears the field, the field is reinitialized with the `missing` value.
    This would imply a required field.

`default` only is declared
:   The block form is initialized with the `default` value.
    When the user clears the field, the field remains empty.
    This would imply an optional field.

`missing` only is declared
:   The block form is initialized with the `missing` value.
    When the user clears the field, the field is reinitialized with the `missing` value.

`default` and `missing` are not declared
:   The block form has no initialized value.
    The field is required.
```

Are "block form" and "field" the same thing or different? In my brain, a form is a collection of one or more fields (a schema) with additional attributes.

Also I am struggling with the term "reinitialized". Does that mean the value is shown to the user in the field and/or its HTML attribute `value` gets set to the `missing` value? If so, that seems weird to me that when a user clears a field, then its value is reinitialized. Maybe the value is what gets processed outside the client browser? Please help me understand.